client/components/theme/index.jsx

```diff
@@ -233,8 +243,49 @@ export class Theme extends Component {
 		);
 	};
 
+	goToCheckout = () => {
+		const { siteSlug, eligibleForProPlan } = this.props;
+		const plan = eligibleForProPlan && isEnabled( 'plans/pro-plan' ) ? 'pro' : 'premium';
+
+		if ( siteSlug ) {
+			const params = new URLSearchParams();
+			params.append( 'redirect_to', window.location.href.replace( window.location.origin, '' ) );
+
+			window.location.href = `/checkout/${ encodeURIComponent(
+				siteSlug
+			) }/${ plan }?${ params.toString() }`;
+		}
+	};
+
+	getUpsellMessage() {
+		const { isPremiumThemesAvaiable, eligibleForProPlan, theme, price, translate } = this.props;
+		const hasPrice = /\d/g.test( price );
+
+		if ( ! hasPrice && theme.price && ! isPremiumThemesAvaiable ) {
+			return translate( 'You have purchased an annual subscription for this theme' );
+		} else if ( isPremiumThemesAvaiable ) {
+			return translate( 'The premium theme is included in your plan.' );
+		}
+
+		return createInterpolateElement(
+			sprintf(
+				/* translators: the "price" is the price of the theme, example: U$50; the "planName" can be "Pro plan" or "Premium plan". */
+				translate(
+					'This premium theme is included in the <Link>%(planName)s</Link>, or you can purchase individually for %(price)s a year'
+				),
+				{
+					price: theme.price,
+					planName: eligibleForProPlan ? translate( 'Pro plan' ) : translate( 'Premium plan' ),
```

Comment:
I don't think we want to check for the Pro plan here now that Premium is live. If someone has the Pro plan, they shouldn't see this upsell, and if they have a free plan, they should see Premium.

Reply:
Should we say "Premium or Business plan" instead of "Premium plan"?

Reply:
Let's stick with just "Premium plan" and not complicate the user messaging too much.